rc/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+use std::ops::Deref;
+use std::ptr;
+
+pub struct MyRc<T> {
+    ptr: *mut T,
+    count: *mut usize,
+}
+
+impl<T> Deref for MyRc<T> {
+    type Target = T;
+
+    fn deref(&self) -> &T {
+        unsafe { & *self.ptr }
+    }
+}
+
+impl<T> Clone for MyRc<T> {
+    fn clone(&self) -> MyRc<T> {
+        unsafe { *self.count += 1; }
+        MyRc { ptr: self.ptr, count: self.count }
+    }
+}
+
+impl<T> MyRc<T> {
+    fn drop(&mut self) {
+        unsafe {
+            *self.count -= 1;
+
+            if *self.count == 0 {
+                ptr::drop_in_place(self.ptr);
+                ptr::drop_in_place(self.count);
```

Comment:
So while this runs the destructor for the contents, it doesn't actually free the memory (see the documentation for `drop_in_place`).

This leads to memory leaks :frowning_face: